gossip/src/cluster_info.rs

```diff
@@ -3789,6 +3814,44 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_push_duplicate_vote() {
+        solana_logger::setup();
+        let mut rng = rand::thread_rng();
+        let keys = Keypair::new();
+        let contact_info = ContactInfo::new_localhost(&keys.pubkey(), 0);
+        let cluster_info = ClusterInfo::new(
+            contact_info,
+            Arc::new(Keypair::new()),
+            SocketAddrSpace::Unspecified,
+        );
+
+        // Add a vote
+        let tx = new_vote_transaction(&mut rng, vec![7, 8, 9]);
+        let tower = vec![1, 2, 3, 4, 5, 6, 7, 8, 9];
+        cluster_info.push_vote(&tower, tx.clone());
+
+        // Add a duplicate vote for the same slots
+        let dup_tx = new_vote_transaction(&mut rng, vec![7, 8, 9]);
+        cluster_info.push_vote(&tower, dup_tx);
+
+        // Make sure we only have the old tx
+        let mut cursor = Cursor::default();
+        let (labels, votes) = cluster_info.get_votes_with_labels(&mut cursor);
+        assert_eq!(votes, vec![tx.clone()]);
+        assert_eq!(labels.len(), 1);
+
+        // Add a duplicate vote for the same slot, but with a different diff
+        let dup_tx2 = new_vote_transaction(&mut rng, vec![9]);
+        cluster_info.push_vote(&tower, dup_tx2);
+
+        // Make sure we only have the original tx
+        cursor = Cursor::default();
+        let (labels, votes) = cluster_info.get_votes_with_labels(&mut cursor);
+        assert_eq!(votes, vec![tx]);
+        assert_eq!(labels.len(), 1);
+    }
```

Comment:
Maybe an extra case here to see that once we push the rest of the 30 votes, the tower doesn't panic